src/main/java/kr/hhplus/be/server/application/coupon/CouponService.java

```diff
@@ -52,17 +68,26 @@ public Long calculateDiscountPrice(Long userCouponId, Long totalPrice) {
             backoff = @Backoff(delay = 10)
     )
     public UserCoupon issueCoupon(Long userId, Long couponId) {
-        Coupon coupon = findCouponById(couponId);
-
-        if (coupon.getStock() <= 0) {
-            throw new OutOfStockCouponException("쿠폰 재고가 부족합니다.");
-        }
-
-        coupon.decreaseStock();
-        couponRepository.save(coupon);
-        
-        UserCoupon userCoupon = UserCoupon.of(userId, couponId);
-        return userCouponRepository.save(userCoupon);
+        // 쿠폰별 분산락 적용
+        return distributedLockService.executeWithLock("coupon:issue:" + couponId, 10, 30, () -> {
```

Comment:
```suggestion
        return distributedLockService.executeWithLock(LockKeyManager.getCouponLockKey(couponId), 10, 30, () -> {
```

하드코딩된 부분을 개선하면 좋을거같아요 뒤에 10, 30 과 같은 숫자들도 변수로 표현해주면 좋겠네요!

Reply:
여담이지만 쿠폰이 중복 발급에 대해서 예외처리도 있으면 좋을것 같네요!